### DISCUSSION TITLE
Adding additional column data to pivot table when attaching records

### DISCUSSION THREAD

**Discussion by richtarr** [filamentphp/filament#103] [Help]
Hi guys - great job on this package, really impressed with how easy it is to get up and running!

The only major thing I'm missing at the minute is the ability to add data into additional columns on a pivot table. For example, I have a BelongsToMany relationship between 'Quote' and 'Item' models, and need to be able to enter data into a 'quantity' column on the intermediate table.

Is this possible currently? I've added the quantity field to the form schema, but this only appears on creating a new item relationship - it feels like there should be an option for it to appear here:

<img width="826" alt="Screenshot 2021-03-06 at 20 27 26" src="https://user-images.githubusercontent.com/6196331/110219963-5fba6500-7eba-11eb-8425-d0682da63e77.png">

Thanks again for the great work!

**Comment by ryangjchandler** [comment]
Hi @richtarr, this is something that we've thought about and I'll look into this week probably. 

I imagine the best way to do it would be adding a `pivotForm` method to the relation manager where you can add all of your fields. What do you think?

**Reply by richtarr** [reply]
Hi @ryangjchandler - yep that would make sense, sounds like the most elegant solution to me!

Thanks very much

**Reply by richtarr** [reply]
Hi @ryangjchandler - any update on this one please?

Happy to have a crack at it if you think it will be a way off, might be a little beyond me though...!

Thanks

**Comment by skoontastic** [comment]
Yep, something like this would be awesome when creating or attaching relations with pivot tables:

![filament-attach-with-pivot-modal](https://user-images.githubusercontent.com/585102/117359276-c76e3b80-ae7c-11eb-80d8-adba95d4aed2.png)
![filament-create-with-pivot-modal](https://user-images.githubusercontent.com/585102/117359279-c76e3b80-ae7c-11eb-86e4-5281ce68b365.png)

**Comment by titantwentyone** [comment]
Also interested in this. Was having a look at the source and wondered if it was possible/advisable to override the `RelationManager\AttachRecord::class` given in `RelationManager`? My thought was to extend this class and modify the `form` method.

in `RelationManager`
```
    protected static $components = [
        'attach' => RelationManager\AttachRecord::class,
        'create' => RelationManager\CreateRecord::class,
        'edit' => RelationManager\EditRecord::class,
        'list' => RelationManager\ListRecords::class,
    ];
```
in `AttachRecord`:
```
    protected function form(Form $form)
    {
        return $form
            ->schema([
                Select::make('related')
                    ->label((string) Str::of($this->getRelationshipName())->singular()->ucfirst())
                    ->placeholder('filament::resources/relation-manager.modals.attach.form.related.placeholder')
                    ->getOptionSearchResultsUsing(function ($search) {
                        $relationship = $this->owner->{$this->getRelationshipName()}();

                        $query = $relationship->getRelated();

                        $search = Str::lower($search);
                        $searchOperator = [
                            'pgsql' => 'ilike',
                        ][$query->getConnection()->getDriverName()] ?? 'like';

                        return $query
                            ->where($this->getPrimaryColumn(), $searchOperator, "%{$search}%")
                            ->whereDoesntHave($this->getInverseRelationshipName(), function ($query) {
                                $query->where($this->owner->getQualifiedKeyName(), $this->owner->getKey());
                            })
                            ->pluck($this->getPrimaryColumn(), $query->getKeyName())
                            ->toArray();
                    })
                    ->required(),

                    //can we tag on our pivot table fields here..?
            ]);
    }
```

**Reply by danharrin** [reply]
Yeah you can do that, you need to also save that information when the relationship is attached though.

**Reply by titantwentyone** [reply]
Would that be that possible using `beforeSave` on the main resource?

**Reply by titantwentyone** [reply]
Scrap that - I think we need to override `attach` in `AttachRecord`

**Reply by titantwentyone** [reply]
Works perfectly! Thanks again Dan!

**Reply by danharrin** [reply]
Awesome! I'm glad.

**Reply by underdpt** [reply]
> 
> 
> Works perfectly! Thanks again Dan!

Hello,

Would be great if you can share your solution ;-)

**Comment by danharrin** [comment]
By the way - this has been released in v2 - https://filamentadmin.com/docs/2.x/admin/resources#pivot-attributes
